api_tests/waffle/views/test_waffle_cookies.py

```diff
@@ -0,0 +1,83 @@
+import mock
+import pytest
+import waffle
+
+from osf_tests.factories import (
+    AuthUserFactory,
+    FlagFactory,
+    ProjectFactory
+)
+from website.ember_osf_web.decorators import camouflage_request
+
+from website.project.views.node import _view_project
+from functools import wraps
+from api.base.views import root
+from framework.auth.core import _get_current_user
+from tests.json_api_test_app import JSONAPITestApp
+from tests.base import OsfTestCase
+
+
+class MockUser(object):
+    is_authenticated = False
+
+
+def check_for_test_flag(func):
+    @wraps(func)
+    def wrapped(self, *args, **kwargs):
+        from flask import request
+        request.user = _get_current_user() or MockUser()
+        request = camouflage_request(request)
+        print waffle.flag_is_active(request, 'test_flag')
+        return func(self, *args, **kwargs)
+    return wrapped
+
+
+@pytest.mark.django_db
+class TestWaffleCookies(OsfTestCase):
+
+    def setUp(self):
+        super(TestWaffleCookies, self).setUp()
+        self.flag = FlagFactory(name='test_flag')
+        self.flag.percent = 50
+        self.flag.everyone = None
+        self.flag.save()
+
+    @pytest.mark.enable_bookmark_creation
+    @mock.patch('website.project.views.node._view_project', check_for_test_flag(_view_project))
+    def test_waffle_leaves_cookie(self):
+        """
+        Tests that django-waffle cookies work in the with our Flask requests.
+
+        Flask waffle cookies are formatted:
+        `dwf_test_flag=True; Expires=True; Max-Age=2592000; Path=/`
+
+        """
+        node = ProjectFactory(is_public=True)
+        user = AuthUserFactory()
+        resp = self.app.get(node.web_url_for('view_project'), auth=user.auth, auto_follow=True)
+
+        waffle_cookie, csrf_cookie = [value for key, value in resp.headers.items() if key == 'Set-Cookie']
+        print waffle_cookie
+        cookie_str = 'dwf_test_flag={}; Expires=True; Max-Age=2592000; Path=/'
+
+        assert waffle_cookie == cookie_str.format('True') or waffle_cookie == cookie_str.format('False')
+        assert 'api-csrf=' in csrf_cookie
+
+    @pytest.mark.enable_bookmark_creation
+    @mock.patch('api.base.views.root', check_for_test_flag(root))
```

Comment:
Even though we check all flags in the DB for the root I want to use a test flag here because I don't want migrations/admin action to cause this to fail.

Reply:
1. It's important to test that the cookie will be set when the ember app hits `/v2/` in real life. The decorator here means we're not testing that at all.
2. `/v2/` is a django endpoint, not flask. It should not use `camouflage_request`.
3. If you only assert that the `test_flag` cookie is set, the presence of other flags/cookies shouldn't interfere.

Reply:
yes I understand better now and fixed this.